### PR TITLE
Remove dependency on "stuff"

### PR DESCRIPTION
**Pull request by Xiaogrill**
Straightforward replacement of `ifconfig` with `ip`.

### CODE REVIEW

**Comment by Xiaogrill** [comment]
Also added "Remove dependency on awk".

**Comment by Xiaogrill** [comment]
Working on removing `wireless-tools` as dependency since it only gets used in dead code.

**Comment by Xiaogrill** [comment]
Actually using the open WiFi capability of https://github.com/br0ns/net/pull/10/commits/42c412c77280e4e285da061723e83da429531801 is _untested_ since I do not have any way to test it here right now.

File: net [hunk 1]
```diff
@@ -1113,21 +1113,23 @@ def do_connect(conn, psk=None):
...
-            else:
-                cmd += ' essid any'
-            run(cmd)
+    if not wpa and not is_wifi_iface(iface):
         if not addr:
             print 'Sending DHCP request'
         for cmd in network:
             run_or_die(cmd)
     else:
+        if not wpa:
```
Comment (TethysSvensson) [inline net]:
```
else:
    if ...:
```

Why not

```
elif ...:
```

?

Comment (Xiaogrill) [reply net]: Falling through to wpa.
The logic overall is shitty--not only in regards to else/elif.

Comment (Xiaogrill) [reply net]: Added clarifying comments in https://github.com/br0ns/net/pull/10/commits/d49dede98c41ede2cc25502a9d0d1c53f2088b07 .

**Comment by Xiaogrill** [comment]
Everything is tested now, ready to merge.

File: net [hunk 2]
```diff
@@ -1113,21 +1113,27 @@ def do_connect(conn, psk=None):
...
+    if not is_wifi_iface(iface):
+        if is_wifi:
+            die('WiFi options set on wired interface!')
         if not addr:
             print 'Sending DHCP request'
         for cmd in network:
             run_or_die(cmd)
+    # Else WiFi
     else:
+        # If Open WiFi with fallthrough--always use wpa_supplicant
```
Comment (TethysSvensson) [inline net:1126]: Why do you want to use wpa_supplicant for open networks?

Comment (Xiaogrill) [reply net:1126]: Got rid of `iwconfig`, and easy fix versus changed flow to connect with `iw`--which might be a fix for the future :)
Also; contemplating removing dependency on `iw` and only use `wpasupplicant`.

**Pull request merged**